### PR TITLE
Functional Test the CGID Feature

### PR DESCRIPTION
**Pull request by adityaguruprasad**
## Description

Functional test the CGID Feature:
Test 1: FPID from the identityMap is used to generate an ECID
Test 2: FPID from a FPID cookie is used to generate an ECID
Test 3: Existing identity cookie takes precedence over an FPID provided in the identity map

## Related Issue
ECID / FPID features not yet included in testing.

## Motivation and Context
Testing CGID features.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Improvement (non-breaking change which does not add functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html) or I'm an Adobe employee.
- [x] I have made any necessary test changes and all tests pass.
- [x] I have run the Sandbox successfully.

### CODE REVIEW

File: test/functional/helpers/cookies.js [hunk 1]
```diff
@@ -11,6 +11,14 @@ export default {
       document.cookie = `${name}=;expires=Thu, 01 Jan 1970 00:00:00 GMT`;
     }
   }),
+  set: ClientFunction(options => {
+    const { name, value, duration = 3600 } = options;
+    const durationMs = duration * 1000;
+    const expires = new Date(Date.now() + durationMs);
+    document.cookie = `${name}=${escape(
+      value
+    )}; expires=${expires.toUTCString()}; domain=".alloyio.com"`;
```
Comment (liljenback) [inline test/functional/helpers/cookies.js:20]: Maybe import the `alloyio.com` domain from a the [domains.js](https://github.com/adobe/alloy/blob/main/test/functional/helpers/constants/domain.js) constants.

Comment (jonsnyder) [reply test/functional/helpers/cookies.js:20]: I don't think you need the leading dot on the domain name.

File: test/functional/specs/Identity/cgid_FT2.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  await alloy.configure(config);
+  await cookies.set({
+    name: "fpidCookie",
+    value: uuid()
+  });
+
+  await alloy.sendEvent(cookies.value);
+  const ecid = getReturnedEcid(networkLogger.edgeEndpointLogs.requests[0]);
+  await cookies.remove(MAIN_IDENTITY_COOKIE_NAME);
+  await reloadPage();
```
Comment (jonsnyder) [inline test/functional/specs/Identity/cgid_FT2.js:63]: The cookie needs to be removed after the reload for it to actually remove the cookie.

File: test/functional/specs/Identity/cgid_FT2.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+  SEVERTIY: "P0",
+  TEST_RUN: "Regression"
+});
+
+test("cgif_FT2: FPID from FPID cookie generates ECID", async () => {
+  const alloy = createAlloyProxy();
+
+  await alloy.configure(config);
+  await cookies.set({
+    name: "fpidCookie",
```
Comment (jonsnyder) [inline test/functional/specs/Identity/cgid_FT2.js:56]: I think this needs to be "FPID". You can have a custom cookie name, but it has to be configured in the datastream.

File: test/functional/specs/Identity/cgid_FT2.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+    value: uuid()
+  });
+
+  await alloy.sendEvent(cookies.value);
```
Comment (jonsnyder) [inline test/functional/specs/Identity/cgid_FT2.js:60]: No need to pass anything into this sendEvent or the next one.

**Review by jonsnyder — CHANGES_REQUESTED** [review]

**Pull request closed without merging**